config/initializers/initialization_cachebust.rb

```diff
@@ -0,0 +1,4 @@
+# Trigger cache purges for globally-cached endpoints that could have changed
```

Comment:
this part I'm not sure I understand. Why do we need to bust the cache at each boot up if we have the version thing? Can't we just not cache `/async_info/shell_version` ?

There's also another issue: boot up is already slow and we're adding 6 HTTP calls (each bust is 2 calls). If they are needed, can they be done async?

Reply:
Yeah, they can be done async.

I think the shell version endpoint should be cached because it will be called millions of times per week and changes infrequently.

It's basically a static endpoint.

I feel like this isn't the kind of HTTP call which will cause us a lot of problems because they're designed to be efficient and highly available. But yeah, we definitely could do this async if you think that is best. But the process of spinning up the delayed job seems kind of just as resource intensive to me in a way.

.........Now that I think about it........ I think the two calls to the API per cachebust call is a hack meant to resolve some old old race conditions which I don't think affect us anymore, so I think we could remove that second purge altogether. 🙃

Reply:
I vote for async and revisiting how we do cache busting altogether in the future. One thing at a time :D 

The drawback about doing them async is that we don't really use priority queues so those two jobs might not be processed quickly when the delayed job workers boot up after the restart